### PR TITLE
[json-rpc] Avoid extra get_currencies lookup for get_account

### PR DESCRIPTION
**Pull request by phlip9**
## Overview

**Problem:** In order to project an `AccountState` into a json-rpc `AccountView`, we need to pull out the balance resources from the `AccountState`, which currently requires a list of currency code `Identifiers` (e.g. "XUS" or "XDX") to build the access `Path` needed to lookup the resource bytes from the state. To get the list of currency codes, we first need to lookup the global `RegisteredCurrencies` set, which requires an extra request for the verifying client's `get_account` implementation and doubles the read-amplification for normal `get_account` requests.

**Solution:** Instead of indexing by currency code to get the balance resources, this diff just reads all balance resources directly and extracts the currency code from the generic type parameter. That way, we don't need to first lookup the global supported currencies list. Likewise for preburn and preburn queue resources.

###  [types] Get balances from an account state without indexing by currency

The previous version required callers to provide a list of currency codes in order to retrieve the `BalanceResource`s in an `AccountState`. With this change, we instead look up all `BalanceResource`s under an `AccountState` directly (there may be several with different generic coin type parameters).

Likewise, we simply list all `PreburnResource`s and `PreburnQueueResource`s instead of indexing by currency codes.

This change will allow us to avoid looking up the set of registered currencies from the `DiemRoot` account before we're able to deserialize account balances from an `AccountState`.

### [json-rpc] Avoid extra get_currencies lookup for get_account

Following the previous commit, we can avoid doing an extra `get_currencies` call in `get_account` since we no longer need the list of currencies to index into the current balance resources. The same is true for the verifying client's implementation of `get_account`. This halves the number of requests needed to support the verifying client's `get_account` call.

### CODE REVIEW

**Comment by libra-action** [comment]
**This PR may have modified JSON-RPC server code.**

Breaking changes policy:

1. Chaning JSON-RPC method signatures or removing/modifying fields
   in the response in /V1 will not be accepted.
2. Adding fields is ok.
3. Adding a new JSON-RPC method is ok.

Please ensure you have also done the following to update the docs:

1. Update json-rpc/API-CHANGELOG.md.
2. Add/update type/method documentation under json-rpc/docs.

**Review by xli — APPROVED** [review]

File: json-rpc/src/data.rs [hunk 1]
```diff
@@ -87,7 +84,7 @@ pub fn get_metadata(
 /// Returns account state (AccountView) by given address
 pub fn get_account(
     db: &dyn DbReader,
-    ledger_version: u64,
+    _ledger_version: u64,
```
Comment (xli) [inline json-rpc/src/data.rs:87]: should we remove this?

**Review by tzakian — APPROVED** [review]
This is awesome! It's so nice that you're able to do this now that we're no longer hashing the access paths :)

**Comment by phlip9** [comment]
rebase removed an import...

/land

**Pull request merged**